lib/gsockaddr.c

```diff
@@ -116,6 +130,42 @@ g_sockaddr_set_port(GSockAddr *a, guint16 port)
   return a->sa_funcs->set_port(a, port);
 }
 
+guint8 *
+g_sockaddr_get_address(GSockAddr *self, guint8 *buffer, socklen_t buffer_size)
+{
+  if (self->sa.sa_family == AF_INET)
+    {
+      struct in_addr addr = g_sockaddr_inet_get_address(self);
+      socklen_t len = sizeof(addr);
+      if (buffer_size < len)
+        {
+          errno = ENOSPC;
```

Comment:
ENOSPC: No space left on device. I think EINVAL would be better.

Reply:
Fixed. Thanks.